src/popup/utils/utils.ts

```diff
@@ -33,28 +41,56 @@ export function formatTime(time: number) {
   });
 }
 
-export function checkHashType(fullHash: string) {
+export function getHashPrefix(fullHash: string) {
   const addressTypes: Record<string, string> = {
-    ak: 'account/transactions',
-    ct: 'contracts/transactions',
-    nm: 'names',
-    ok: 'oracles/queries',
-    th: 'transactions',
+    [HASH_PREFIX_ACCOUNT]: 'account/transactions',
+    [HASH_PREFIX_CONTRACT]: 'contracts/transactions',
+    [HASH_PREFIX_NAME]: 'names',
+    [HASH_PREFIX_ORACLE]: 'oracles/queries',
+    [HASH_PREFIX_TRANSACTION]: 'transactions',
   };
+
   const [prefix, hash] = fullHash.split('_');
 
-  let valid: boolean = false;
-  let endpoint: string | null = null;
+  return {
+    hash,
+    prefix,
+    addressTypes,
+  };
+}
 
-  if (fullHash.endsWith('.chain')) {
-    valid = true;
-    endpoint = addressTypes.nm;
-  } else if (addressTypes[prefix] && HASH_REGEX.test(hash)) {
-    valid = true;
-    endpoint = addressTypes[prefix];
+export function validateHash(fullHash?: string) {
+  if (!fullHash) {
+    return {};
   }
+  const { addressTypes, prefix, hash } = getHashPrefix(fullHash);
+
+  const isNameHash = fullHash.endsWith('.chain');
+  const valid: boolean = (addressTypes[prefix] && HASH_REGEX.test(hash)) || isNameHash;
+
+  return {
+    valid, isNameHash, addressTypes, prefix,
```

Comment:
As this function name is `validate` in my opinion it should only return bool and optionally the prefix. No more data.